dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ArgumentTypesResolver.java

```diff
@@ -97,10 +97,15 @@ private Optional<ArgumentType> getPrimitiveType(TypeMirror typeMirror) {
 
     private Optional<ArgumentType> getListType(TypeMirror typeMirror) {
         TypeName typeName = TypeName.get(typeMirror);
-
         return context.getGenericInnerType(List.class, typeMirror)
-                .flatMap(this::getPrimitiveSerializerMethodName)
-                .map(methodName -> ArgumentTypes.list(typeName, methodName));
+                .flatMap(this::getListInnerType)
+                .map(argumentType -> ArgumentTypes.list(
+                        typeName,
+                        ImmutableListType.builder().innerType(argumentType).build()));
+    }
+
+    private Optional<ArgumentType> getListInnerType(TypeMirror typeMirror) {
```

Comment:
nit: Inline this - it's only used in one place and inlining is consistent with the design of `getOptionalType` below